compliance/collection/main.go

```diff
@@ -182,17 +217,106 @@ func manageNodeScanLoop(ctx context.Context, rescanInterval time.Duration, scann
 	return sensorC
 }
 
-func scanNode(nodeName string, scanner nodeinventorizer.NodeInventorizer) (*sensor.MsgFromCompliance, error) {
-	result, err := scanner.Scan(nodeName)
+func runCachedScan(opts *cachedScanOpts) (*storage.NodeInventory, error) {
```

Comment:
I think, `runScanAndCacheResults` would be a better name. Again, this means that this function mixes two concerns and should better be split. That said, right now it does not look that bad so I don't insist on splitting that.

Reply:
I've revisited all function names in the code reorganization, done!